src/modules/pageTable.js

```diff
@@ -119,23 +175,201 @@ class PageTable extends React.PureComponent {
             </div>
         );
     }
+
+    _onSplitterClick() {
+        const { onSplitter } = this.props;
+        onSplitter();
+    }
 }
 
 PageTable.defaultProps = {
     bleed: true,
     fontSize: 'xsmall',
+    idPrefix: 'base',
     small: true,
+    splitter: false,
 };
 
 PageTable.propTypes = {
+    bleed: PropTypes.bool,
+    className: PropTypes.string,
+    collapsed: PropTypes.bool,
+    columns: PropTypes.array.isRequired,
+    data: PropTypes.array.isRequired,
+    fontSize: PropTypes.string,
+    idPrefix: PropTypes.string,
+    onScrollColumn: PropTypes.func,
+    onSplitter: PropTypes.func,
+    rowProps: PropTypes.func,
+    sizes: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.object)),
+    small: PropTypes.bool,
+    splitter: PropTypes.bool,
+    style: PropTypes.object,
+};
+
+class PageTableContainer extends React.Component {
+    constructor(props) {
+        super(props);
+        this._onResize = this._onResize.bind(this);
+        this._onScrollColumn = this._onScrollColumn.bind(this);
+        this._onSplitterClick = this._onSplitterClick.bind(this);
+        this.state = {
+            collapsed: null,
+            minWidth: props.minWidth,
+            sizes: [],
+        };
+    }
+
+    componentDidMount() {
+        this._onResize();
+        window.addEventListener('resize', this._onResize);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this._onResize);
+    }
+
+    render() {
+        const { columns, stickyColumns } = this.props;
+        const { collapsed, minWidth, sizes } = this.state;
+
+        if (stickyColumns > 0) {
+            return (
+                <div className="ui page--table_container">
+                    <div className="page--table_fixed_body">
+                        <PageTable
+                            {...this.props}
+                            collapsed={collapsed}
+                            idPrefix="body"
+                            onScrollColumn={this._onScrollColumn}
+                            ref={ref => this._bodyTable = ref}
+                            sizes={sizes}
+                            style={{ minWidth }}
+                        />
+                    </div>
+                    <div className="page--table_fixed_column">
+                        <PageTable
+                            {...this.props}
+                            columns={_.slice(columns, 0, stickyColumns)}
+                            idPrefix="column"
+                            sizes={sizes}
+                            onSplitter={this._onSplitterClick}
+                        />
+                    </div>
+                </div>
+            );
+        } else {
+            return <PageTable {...this.props} />;
```

Comment:
Let's bring the `<div className="page--table" />` down to this parent component and wrap this `<PageTable />` child component and the above stickyColumns jsx.

`class UITable` -> `class UITable`

```
    return (
        <Table className="page--ui_table">
            ...
        </Table
    );
```

`class PageTableContainer` -> `class PageTable`

```
const isStickyColumns = stickyColumns > 0;
const containerClasses = ClassNames('ui', 'page--table', className, {
    page--table-sticky_columns': isStickyColumns,
});
const newStyle = Object.assign({}, style, {
    margin: bleed ? '0 -22px' : null,
});

if (isStickyColumns) {
    return (
        <div
            className={containerClasses}
            style={newStyle}
        >
           <div className="page--table_fixed_body">
               <PageTable
                   {...this.props}
                   collapsed={collapsed}
                   idPrefix="body"
                   onScrollColumn={this._onScrollColumn}
                   ref={ref => this._bodyTable = ref}
                   sizes={sizes}
                   style={{ minWidth }}
               />
           </div>
           
           <div className="page--table_fixed_column">
               <PageTable
                   {...this.props}
                   columns={_.slice(columns, 0, stickyColumns)}
                   idPrefix="column"
                   sizes={sizes}
                   onSplitter={this._onSplitterClick}
               />
           </div>
        </div>
    );
} else {
   return (
        <div
            className={containerClasses}
            style={newStyle}
        >
            <PageTable {...this.props} />
        </div>
    );
}
```

Reply:
Done.